tests/kernel/timer/timer_api/src/main.c

```diff
@@ -142,6 +142,16 @@ static void status_stop(struct k_timer *timer)
  */
 void test_timer_duration_period(void)
 {
+	init_timer_data();
+	/** TESTPOINT: start timer via k_timer_start */
+	k_timer_start(&duration_timer, K_MSEC(DURATION), K_MSEC(PERIOD));
+	tdata.timestamp = k_uptime_get();
+	/** TESTPOINT: check timer status of be stopped in mid-countdown */
+	busy_wait_ms(DURATION / 2);
```

Comment:
There's a test somewhere that does exactly this already.  I know because it had a bug this cycle where on nRF the busy wait timer is too fast and the "half duration" is a tick short.  Please find that one and add your two assertions to that to avoid adding run time.

Reply:
I have added two assertions to that test and remove my test.

Reply:
I have added two assertions to that test and remove my test.